src/msg/async/AsyncConnection.cc

```diff
@@ -2364,12 +2358,19 @@ ssize_t AsyncConnection::write_message(Message *m, bufferlist& bl)
                              << " data=" << header.data_len
                              << " off " << header.data_off << dendl;
 
-  complete_bl.claim_append(bl);
+  if (bl.length() <= 256) {
```

Comment:
I agree, this would be better if done on upper layers, but that calls for far more changes than this and in a lot of different places. When done here, we'll be sure that we're actually reducing number of iovecs pointing to small buffers, and with little luck, we'll end up with one large buffer (instead of three on average), and we'll save time normally spent on proceeding to other iovec and memmove'ing it. In other words, while still need to iterate over bl (which has N elements), we don't ask sendmsg to iterate on N more elements.